### ISSUE TITLE
Add docs about FanoutCache shard size limit and "none" eviction policy

### ISSUE THREAD

**Issue by eromoe** [grantjenks/python-diskcache#105]
Hello,
I have a function need cache a large pandas DataFrame loaded from s3 . 
The file size would be 518MB by flask-caching (and 826MB by joblib ).

I don't want to cache the flask view, also want to reuse this cache file in some daily jobs. So I switch to diskcache. 

But I found diskcache didn't work .
```
cache = FanoutCache(CACHE_DIR, shards=4, timeout=20)

@cache.memoize()
def read_saleinfo():
    reader = get_reader(DATA_SOURCE)
    days = pd.date_range(start=FORECAST_START_DATE, end=date.today(), freq='D').to_series().apply(lambda x: x.strftime('%Y/%m/%d')).ravel()
    p1 = [SOURCE_DIR_PATH + "arch/test/%s/*.parquet" % i for i in days]
    df = reader.read_paths(p1, columns=['product_id', 'store_id', 'count_of_sales', 'price_of_sales', 'price_max', 'year', 'month', 'day'])

    return df
```

test code: 
```
df1 = read_saleinfo()
df2 = read_saleinfo()
```

`read_saleinfo` actually execute twice .

Do I miss something ?

**Comment by eromoe** [comment]
It worked after I change `size_limit=10 ** 30` . 
Dump file size is 518MB , it must have some problem , at least no error message output in my situation.

**Comment by grantjenks** [comment]
Your FanoutCache uses four shards and default size limit is 1 GB so each shard can hold about 256 MB before evicting items. In your case, the item is evicted immediately due to its large size.

A word of caution: The size limit 10 ** 30 is effectively infinite. That’s more than a petabyte. Try 2**40 for a terabyte.

You can also choose the “none” eviction policy and it will never evict keys.

**Comment by grantjenks** [comment]
Docs improvement ideas:

1. Add note about shard size limit for FanoutCache 
2. Add “none” eviction policy to tutorial docs

**Comment by grantjenks** [comment]
How would that have helped you?

**Comment by eromoe** [comment]
I already known. But I think that message can help someone who come in this situation in future.

**Comment by eromoe** [comment]
Oh, I got another problem : diskcache would save settings in sqlite, even if the settings is wrong .

I thought "none" eviction policy you mentioned was already existed but only not wrote in docs. 

Change code to 
```
cache = FanoutCache(CACHE_DIR, shards=4, timeout=20, eviction_policy=None)
```
got error. Change back, still same error : 

```
Traceback (most recent call last):
  File "index.py", line 10, in <module>
    from src import order_source_statistics
  File "E:\pp\bihome\src\order_source_statistics.py", line 17, in <module>
    from mlc.common import get_main_db
  File "e:\pp\mlc\mlc\common.py", line 22, in <module>
    cache = FanoutCache(CACHE_DIR, shards=4, timeout=20, size_limit=2**40)
  File "C:\Anaconda3\lib\site-packages\diskcache\fanout.py", line 38, in __init__
    for num in range(shards)
  File "C:\Anaconda3\lib\site-packages\diskcache\fanout.py", line 38, in <genexpr>
    for num in range(shards)
  File "C:\Anaconda3\lib\site-packages\diskcache\core.py", line 467, in __init__
    query = EVICTION_POLICY[self.eviction_policy]['init']
KeyError: None
```

**Comment by grantjenks** [comment]
The policy is ``'none'`` not ``None``. It's in the docs at: http://www.grantjenks.com/docs/diskcache/api.html#diskcache.EVICTION_POLICY

**Comment by eromoe** [comment]
Sorry, I miss that. Here only wrote three http://www.grantjenks.com/docs/diskcache/tutorial.html#eviction-policies ，so I didn't look deep.

**Comment by grantjenks** [comment]
Fixed in v4-docs branch at 07f6e5f and 71d1b47.

Thank you for reporting these docs issues.
